src/wp-includes/media.php

```diff
@@ -5449,3 +5449,283 @@ function wp_increase_content_media_count( $amount = 1 ) {
 
 	return $content_media_count;
 }
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @since 1.2.0
+ *
+ * @param array $metadata      The attachment metadata.
+ * @param int   $attachment_id The attachment ID.
+ * @return array $metadata The attachment metadata.
+ */
+function dominant_color_metadata( $metadata, $attachment_id ) {
+	$dominant_color_data = _dominant_color_get_dominant_color_data( $attachment_id );
+	if ( ! is_wp_error( $dominant_color_data ) ) {
+		if ( isset( $dominant_color_data['dominant_color'] ) ) {
+			$metadata['dominant_color'] = $dominant_color_data['dominant_color'];
+		}
+
+		if ( isset( $dominant_color_data['has_transparency'] ) ) {
+			$metadata['has_transparency'] = $dominant_color_data['has_transparency'];
+		}
+	}
+
+	return $metadata;
+}
+add_filter( 'wp_generate_attachment_metadata', 'dominant_color_metadata', 10, 2 );
+
+/**
+ * Filter various image attributes to add the dominant color to the image
+ *
+ * @since 1.2.0
+ *
+ * @param array  $attr       Attributes for the image markup.
+ * @param object $attachment Image attachment post.
+ * @return mixed $attr Attributes for the image markup.
+ */
+function dominant_color_update_attachment_image_attributes( $attr, $attachment ) {
```

Comment:
We can add these directly herer - https://github.com/WordPress/wordpress-develop/blob/f52d476bc6dcfe4c5d011f335361500f80708620/src/wp-includes/media.php#L1083

Reply:
filter removed and calling function directly